build/gulpfile.vscode.darwin.js

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+const gulp = require('gulp');
+const { signAsync } = require('electron-osx-sign');
+const util = require('./lib/util');
+const path = require('path');
+
+gulp.task('vscode-codesign', async () => {
+	const buildDir = process.env['AGENT_BUILDDIRECTORY'];
+
+	if (!buildDir) {
+		throw new Error('$AGENT_BUILDDIRECTORY not set');
+	}
+
+	if (process.env.VSCODE_QUALITY !== 'insider' && process.env.VSCODE_QUALITY !== 'stable') {
```

Comment:
We also have `exploration`, and potentially more in the future, so I'd like not to lock us down here. Could we simply avoid this check and derivate the app name from the fs?